File: go/sequential/sequential1.go
package sequential

import (
	"fmt"
	"time"

	"github.com/MilicaPoparic/ntp/go/util"
)

func Sequential1(a [][]int, b [][]int, size int) {
	util.WriteMatrix("sequential.txt", "Matrices A, B", a, b)
	startTime := time.Now()
	var c [][]int
	// var c2 [][]int
	for i := 0; i < size; i++ {
		zeros := make([]int, size)
		c = append(c, zeros)
	}
	a, b = util.StepOne(a, b, size)

	c = util.AddAndMultiply(a, b, c, size)
	util.WriteToFile("sequential.txt", 1, a, b, c)

	for i := 1; i < size; i++ {
		for j := 0; j < size; j++ {
			a[j] = append(a[j][1:], a[j][:1]...)
		}
		b = append(b[1:], b[:1]...)
		c = util.AddAndMultiply(a, b, c, size)
		util.WriteToFile("sequential.txt", i+1, a, b, c)
	}

	elapsed := time.Since(startTime)
	for _, row := range c {
		fmt.Println(row)
	}
	fmt.Print("Process finished in ", elapsed)
}

File: python/sequential1.py
import time
from util import add_and_multiply, step_one, write_to_file, write
import numpy

def sequential(a, b, size):
    start_time = time.time()
    write("sequential.txt", "Matrices A, B", a, b)
    c = [[0 for row in range(size)] for row in range(size)]
    a, b = step_one(a, b, size)
    c = add_and_multiply(a, b, c, size)
    write_to_file('sequential.txt', 1, a, b, c)

    # other steps (transformations)
    for j in range(1, size):
        for i in range(0, size):
            a[i] = a[i][1:] + a[i][:1]
        b = b[1:] + b[:1]
        c = add_and_multiply(a, b, c, size)

        write_to_file('sequential.txt', j + 1, a, b, c)
    end_time = time.time()
    time1 = end_time - start_time

    print("_______________________")
    for row in c:
        print(row)

    print('Process finished in: ', time1)